### PR TITLE
Add JMX metrics to thread-per-driver scheduler

### PR DESCRIPTION
**Pull request by hashhar**
## Description

Tries to expose the same information available through log diagnostics added in https://github.com/trinodb/trino/pull/23045 over JMX.

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/execution/executor/dedicated/ThreadPerDriverTaskExecutor.java [hunk 1]
```diff
@@ -246,4 +249,67 @@ public Set<TaskId> getStuckSplitTaskIds(Duration processingDurationThreshold, Pr
         // TODO
         return ImmutableSet.of();
     }
+
+    //
```
Comment (wendigo) [inline core/trino-main/src/main/java/io/trino/execution/executor/dedicated/ThreadPerDriverTaskExecutor.java:253]: Drop comment

File: core/trino-main/src/main/java/io/trino/execution/executor/dedicated/ThreadPerDriverTaskExecutor.java [hunk 2]
```diff
@@ -246,4 +249,67 @@ public Set<TaskId> getStuckSplitTaskIds(Duration processingDurationThreshold, Pr
...
+    {
+        return tasks.values().stream()
+                .mapToInt(TaskEntry::totalRunningSplits)
+                .sum();
+    }
+
+    @Managed
+    public synchronized int getTotalRunningLeafSplits()
+    {
+        return tasks.values().stream()
```
Comment (wendigo) [inline core/trino-main/src/main/java/io/trino/execution/executor/dedicated/ThreadPerDriverTaskExecutor.java:273]: do we need synchronization here?

File: core/trino-main/src/main/java/io/trino/execution/executor/dedicated/ThreadPerDriverTaskExecutor.java [hunk 3]
```diff
@@ -246,4 +249,67 @@ public Set<TaskId> getStuckSplitTaskIds(Duration processingDurationThreshold, Pr
...
+    @Managed
+    public synchronized int getTotalPendingLeafSplitCount()
+    {
+        return tasks.values().stream()
+                .mapToInt(TaskEntry::pendingLeafSplitCount)
+                .sum();
+    }
+
+    //
+    // FairScheduler STATS
```
Comment (wendigo) [inline core/trino-main/src/main/java/io/trino/execution/executor/dedicated/ThreadPerDriverTaskExecutor.java:287]: Drop comment

File: core/trino-main/src/main/java/io/trino/execution/executor/dedicated/ThreadPerDriverTaskExecutor.java [hunk 4]
```diff
@@ -246,4 +249,67 @@ public Set<TaskId> getStuckSplitTaskIds(Duration processingDurationThreshold, Pr
...
+    }
+
+    @Managed(description = "Task executor")
+    @Nested
+    public ThreadPoolExecutorMBean getTaskExecutor()
+    {
+        return scheduler.getTaskExecutor();
+    }
+
+    // Add @Managed mbeans for concurreny control total slots, available slots, pending reservations and current reservations.
```
Comment (wendigo) [inline core/trino-main/src/main/java/io/trino/execution/executor/dedicated/ThreadPerDriverTaskExecutor.java:303]: is this a TODO?

File: core/trino-main/src/main/java/io/trino/execution/executor/scheduler/FairScheduler.java [hunk 5]
```diff
@@ -336,4 +341,27 @@ public String toString()
                 .add("closed=" + closed)
                 .toString();
     }
+
+    //
```
Comment (wendigo) [inline core/trino-main/src/main/java/io/trino/execution/executor/scheduler/FairScheduler.java:345]: Drop comment

**Review by wendigo — COMMENTED** [review]
Overall LGTM % comment

This is useful!

**Review by wendigo — APPROVED** [review]

File: core/trino-main/src/main/java/io/trino/execution/executor/dedicated/ThreadPerDriverTaskExecutor.java [hunk 6]
```diff
@@ -246,4 +249,67 @@ public Set<TaskId> getStuckSplitTaskIds(Duration processingDurationThreshold, Pr
...
+    @Managed
+    public synchronized int getTotalRunningLeafSplits()
+    {
+        return tasks.values().stream()
+                .mapToInt(TaskEntry::runningLeafSplits)
+                .sum();
+    }
+
+    @Managed
+    public synchronized int getTotalPendingLeafSplitCount()
```
Comment (martint) [inline core/trino-main/src/main/java/io/trino/execution/executor/dedicated/ThreadPerDriverTaskExecutor.java:279]: getTotalPendingLeafSplits

**Review by martint — APPROVED** [review]

File: core/trino-main/src/main/java/io/trino/execution/executor/dedicated/ThreadPerDriverTaskExecutor.java (hunk 2 above)
Comment (hashhar) [reply core/trino-main/src/main/java/io/trino/execution/executor/dedicated/ThreadPerDriverTaskExecutor.java:273]: Yes, access to `tasks` needs to be synchronized (@GuardedBy("this")).

**Comment by hashhar** [comment]
Addressed comments, will merge once CI is done.

**Comment by hashhar** [comment]
alluxio startup failures: 

```
2024-12-06 12:12:44,248 ERROR [metrics-master-heartbeat-0](SamplingLogger.java:360) - Failed to get master address during initialization
alluxio.exception.status.UnavailableException: Failed to handshake with master localhost:19998 to load cluster default configuration values: UNAVAILABLE: io exception
	at alluxio.conf.Configuration.loadConfiguration(Configuration.java:458)
	at alluxio.ClientContext.loadConf(ClientContext.java:123)
	at alluxio.client.metrics.ClientMasterSync.loadConf(ClientMasterSync.java:122)
	at alluxio.client.metrics.ClientMasterSync.heartbeat(ClientMasterSync.java:74)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)
Caused by: io.grpc.StatusRuntimeException: UNAVAILABLE: io exception
	at io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:271)
	at io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:252)
	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:165)
	at alluxio.grpc.MetaMasterConfigurationServiceGrpc$MetaMasterConfigurationServiceBlockingStub.getConfiguration(MetaMasterConfigurationServiceGrpc.java:405)
	at alluxio.conf.Configuration.loadConfiguration(Configuration.java:452)
	... 10 more
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/127.0.0.1:19998
Caused by: java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:716)
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:330)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:334)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:707)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:655)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:581)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at java.lang.Thread.run(Thread.java:750)
2024-12-06 12:12:49,388 WARN  [scheduler](Scheduler.java:241) - Failed to get worker info, using existing worker infos of 0 workers

Error:  Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 207.0 s <<< FAILURE! -- in io.trino.filesystem.alluxio.TestAlluxioFileSystem
Error:  io.trino.filesystem.alluxio.TestAlluxioFileSystem -- Time elapsed: 207.0 s <<< ERROR!
java.util.concurrent.ExecutionException: org.testcontainers.containers.ContainerLaunchException: Container startup failed for image alluxio/alluxio:2.9.5
	at java.base/java.util.concurrent.CompletableFuture.wrapInExecutionException(CompletableFuture.java:345)
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:440)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2117)
	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:319)
	at org.testcontainers.junit.jupiter.TestcontainersExtension$StoreAdapter.start(TestcontainersExtension.java:276)
	at org.testcontainers.junit.jupiter.TestcontainersExtension$StoreAdapter.access$200(TestcontainersExtension.java:263)
	at org.testcontainers.junit.jupiter.TestcontainersExtension.lambda$null$4(TestcontainersExtension.java:83)
	at org.testcontainers.junit.jupiter.TestcontainersExtension.lambda$startContainers$5(TestcontainersExtension.java:83)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
	at org.testcontainers.junit.jupiter.TestcontainersExtension.startContainers(TestcontainersExtension.java:83)
	at org.testcontainers.junit.jupiter.TestcontainersExtension.beforeAll(TestcontainersExtension.java:57)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:507)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1458)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:2034)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:189)
Caused by: org.testcontainers.containers.ContainerLaunchException: Container startup failed for image alluxio/alluxio:2.9.5
	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:351)
	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:322)
	at java.base/java.util.concurrent.CompletableFuture$UniRun.tryFire(CompletableFuture.java:831)
	at java.base/java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:526)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1575)
Caused by: org.rnorth.ducttape.RetryCountExceededException: Retry limit hit with exception
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:88)
	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:336)
	... 6 more
Caused by: org.testcontainers.containers.ContainerLaunchException: Could not create/start container
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:556)
	at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:346)
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)
	... 7 more
Caused by: org.testcontainers.containers.ContainerLaunchException: Timed out waiting for log output matching '.*Primary started*
'
	at org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy.waitUntilReady(LogMessageWaitStrategy.java:47)
	at org.testcontainers.containers.wait.strategy.AbstractWaitStrategy.waitUntilReady(AbstractWaitStrategy.java:52)
	at org.testcontainers.containers.GenericContainer.waitUntilContainerStarted(GenericContainer.java:909)
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:492)
	... 9 more
```

**Comment by hashhar** [comment]
Iceberg failure - this actually seems like production code issue. cc: @ebyhr can you confirm/deny?

```
Error:  io.trino.plugin.iceberg.TestIcebergV2.testOptimizeDuringWriteOperations -- Time elapsed: 17.30 s <<< ERROR!
java.util.concurrent.ExecutionException: java.lang.RuntimeException: io.trino.testing.QueryFailedException: Failed to commit the transaction during optimize: Cannot commit, found new delete for replaced data file: GenericDataFile{content=data, file_path=local:///tpch/test_optimize_during_write_operationsnedsleh0bt-b45791a5fcb44db595f65eeb1d6b8c67/data/20241206_120318_00236_qmq4f-74b7c592-484f-4d30-8fd8-f36975ec1344.parquet, file_format=PARQUET, spec_id=0, partition=PartitionData{}, record_count=1, file_size_in_bytes=213, column_sizes=null, value_counts=null, null_value_counts=null, nan_value_counts=null, lower_bounds=null, upper_bounds=null, key_metadata=null, split_offsets=[4], equality_ids=null, sort_order_id=0, data_sequence_number=5, file_sequence_number=5}
	at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
	at io.trino.plugin.iceberg.TestIcebergV2.runOptimizeDuringWriteOperations(TestIcebergV2.java:749)
	at io.trino.plugin.iceberg.TestIcebergV2.testOptimizeDuringWriteOperations(TestIcebergV2.java:671)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:507)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.tryRemoveAndExec(ForkJoinPool.java:1499)
	at java.base/java.util.concurrent.ForkJoinPool.helpJoin(ForkJoinPool.java:2272)
	at java.base/java.util.concurrent.ForkJoinTask.awaitDone(ForkJoinTask.java:495)
	at java.base/java.util.concurrent.ForkJoinTask.join(ForkJoinTask.java:662)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:507)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1458)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:2034)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:189)
Caused by: java.lang.RuntimeException: io.trino.testing.QueryFailedException: Failed to commit the transaction during optimize: Cannot commit, found new delete for replaced data file: GenericDataFile{content=data, file_path=local:///tpch/test_optimize_during_write_operationsnedsleh0bt-b45791a5fcb44db595f65eeb1d6b8c67/data/20241206_120318_00236_qmq4f-74b7c592-484f-4d30-8fd8-f36975ec1344.parquet, file_format=PARQUET, spec_id=0, partition=PartitionData{}, record_count=1, file_size_in_bytes=213, column_sizes=null, value_counts=null, null_value_counts=null, nan_value_counts=null, lower_bounds=null, upper_bounds=null, key_metadata=null, split_offsets=[4], equality_ids=null, sort_order_id=0, data_sequence_number=5, file_sequence_number=5}
	at io.trino.plugin.iceberg.TestIcebergV2.lambda$runOptimizeDuringWriteOperations$10(TestIcebergV2.java:733)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1575)
Caused by: io.trino.testing.QueryFailedException: Failed to commit the transaction during optimize: Cannot commit, found new delete for replaced data file: GenericDataFile{content=data, file_path=local:///tpch/test_optimize_during_write_operationsnedsleh0bt-b45791a5fcb44db595f65eeb1d6b8c67/data/20241206_120318_00236_qmq4f-74b7c592-484f-4d30-8fd8-f36975ec1344.parquet, file_format=PARQUET, spec_id=0, partition=PartitionData{}, record_count=1, file_size_in_bytes=213, column_sizes=null, value_counts=null, null_value_counts=null, nan_value_counts=null, lower_bounds=null, upper_bounds=null, key_metadata=null, split_offsets=[4], equality_ids=null, sort_order_id=0, data_sequence_number=5, file_sequence_number=5}
	at io.trino.testing.AbstractTestingTrinoClient.execute(AbstractTestingTrinoClient.java:138)
	at io.trino.testing.DistributedQueryRunner.executeInternal(DistributedQueryRunner.java:565)
	at io.trino.testing.DistributedQueryRunner.executeWithPlan(DistributedQueryRunner.java:554)
	at io.trino.testing.QueryAssertions.assertDistributedUpdate(QueryAssertions.java:106)
	at io.trino.testing.QueryAssertions.assertUpdate(QueryAssertions.java:60)
	at io.trino.testing.AbstractTestQueryFramework.assertUpdate(AbstractTestQueryFramework.java:410)
	at io.trino.testing.AbstractTestQueryFramework.assertUpdate(AbstractTestQueryFramework.java:405)
	at io.trino.plugin.iceberg.TestIcebergV2.lambda$runOptimizeDuringWriteOperations$10(TestIcebergV2.java:730)
	... 5 more
	Suppressed: java.lang.Exception: SQL: ALTER TABLE test_optimize_during_write_operationsnedsleh0bt EXECUTE optimize
		at io.trino.testing.DistributedQueryRunner.executeInternal(DistributedQueryRunner.java:572)
		... 11 more
Caused by: io.trino.spi.TrinoException: Failed to commit the transaction during optimize: Cannot commit, found new delete for replaced data file: GenericDataFile{content=data, file_path=local:///tpch/test_optimize_during_write_operationsnedsleh0bt-b45791a5fcb44db595f65eeb1d6b8c67/data/20241206_120318_00236_qmq4f-74b7c592-484f-4d30-8fd8-f36975ec1344.parquet, file_format=PARQUET, spec_id=0, partition=PartitionData{}, record_count=1, file_size_in_bytes=213, column_sizes=null, value_counts=null, null_value_counts=null, nan_value_counts=null, lower_bounds=null, upper_bounds=null, key_metadata=null, split_offsets=[4], equality_ids=null, sort_order_id=0, data_sequence_number=5, file_sequence_number=5}
	at io.trino.plugin.iceberg.IcebergMetadata.commitTransaction(IcebergMetadata.java:1835)
	at io.trino.plugin.iceberg.IcebergMetadata.commitUpdateAndTransaction(IcebergMetadata.java:1822)
	at io.trino.plugin.iceberg.IcebergMetadata.finishOptimize(IcebergMetadata.java:1790)
	at io.trino.plugin.iceberg.IcebergMetadata.finishTableExecute(IcebergMetadata.java:1711)
	at io.trino.plugin.base.classloader.ClassLoaderSafeConnectorMetadata.finishTableExecute(ClassLoaderSafeConnectorMetadata.java:248)
	at io.trino.tracing.TracingConnectorMetadata.finishTableExecute(TracingConnectorMetadata.java:187)
	at io.trino.metadata.MetadataManager.finishTableExecute(MetadataManager.java:361)
	at io.trino.tracing.TracingMetadata.finishTableExecute(TracingMetadata.java:228)
	at io.trino.sql.planner.LocalExecutionPlanner.lambda$createTableFinisher$4(LocalExecutionPlanner.java:4242)
	at io.trino.operator.TableFinishOperator.getOutput(TableFinishOperator.java:316)
	at io.trino.operator.Driver.processInternal(Driver.java:403)
	at io.trino.operator.Driver.lambda$process$8(Driver.java:306)
	at io.trino.operator.Driver.tryWithLock(Driver.java:709)
	at io.trino.operator.Driver.process(Driver.java:298)
	at io.trino.operator.Driver.processForDuration(Driver.java:269)
	at io.trino.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:890)
	at io.trino.execution.executor.dedicated.SplitProcessor.run(SplitProcessor.java:77)
	at io.trino.execution.executor.dedicated.TaskEntry$VersionEmbedderBridge.lambda$run$0(TaskEntry.java:201)
	at io.trino.$gen.Trino_testversion____20241206_120227_6711.run(Unknown Source)
	at io.trino.execution.executor.dedicated.TaskEntry$VersionEmbedderBridge.run(TaskEntry.java:202)
	at io.trino.execution.executor.scheduler.FairScheduler.runTask(FairScheduler.java:177)
	at io.trino.execution.executor.scheduler.FairScheduler.lambda$submit$0(FairScheduler.java:164)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:131)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:76)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:82)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1575)
Caused by: org.apache.iceberg.exceptions.ValidationException: Cannot commit, found new delete for replaced data file: GenericDataFile{content=data, file_path=local:///tpch/test_optimize_during_write_operationsnedsleh0bt-b45791a5fcb44db595f65eeb1d6b8c67/data/20241206_120318_00236_qmq4f-74b7c592-484f-4d30-8fd8-f36975ec1344.parquet, file_format=PARQUET, spec_id=0, partition=PartitionData{}, record_count=1, file_size_in_bytes=213, column_sizes=null, value_counts=null, null_value_counts=null, nan_value_counts=null, lower_bounds=null, upper_bounds=null, key_metadata=null, split_offsets=[4], equality_ids=null, sort_order_id=0, data_sequence_number=5, file_sequence_number=5}
	at org.apache.iceberg.exceptions.ValidationException.check(ValidationException.java:49)
	at org.apache.iceberg.MergingSnapshotProducer.validateNoNewDeletesForDataFiles(MergingSnapshotProducer.java:493)
	at org.apache.iceberg.MergingSnapshotProducer.validateNoNewDeletesForDataFiles(MergingSnapshotProducer.java:425)
	at org.apache.iceberg.BaseRewriteFiles.validate(BaseRewriteFiles.java:140)
	at org.apache.iceberg.SnapshotProducer.apply(SnapshotProducer.java:252)
	at org.apache.iceberg.SnapshotProducer.lambda$commit$2(SnapshotProducer.java:403)
	at org.apache.iceberg.util.Tasks$Builder.runTaskWithRetry(Tasks.java:413)
	at org.apache.iceberg.util.Tasks$Builder.runSingleThreaded(Tasks.java:219)
	at org.apache.iceberg.util.Tasks$Builder.run(Tasks.java:203)
	at org.apache.iceberg.util.Tasks$Builder.run(Tasks.java:196)
	at org.apache.iceberg.SnapshotProducer.commit(SnapshotProducer.java:401)
	at org.apache.iceberg.BaseTransaction.applyUpdates(BaseTransaction.java:509)
	at org.apache.iceberg.BaseTransaction.lambda$commitSimpleTransaction$3(BaseTransaction.java:425)
	at org.apache.iceberg.util.Tasks$Builder.runTaskWithRetry(Tasks.java:413)
	at org.apache.iceberg.util.Tasks$Builder.runSingleThreaded(Tasks.java:219)
	at org.apache.iceberg.util.Tasks$Builder.run(Tasks.java:203)
	at org.apache.iceberg.util.Tasks$Builder.run(Tasks.java:196)
	at org.apache.iceberg.BaseTransaction.commitSimpleTransaction(BaseTransaction.java:423)
	at org.apache.iceberg.BaseTransaction.commitTransaction(BaseTransaction.java:318)
	at io.trino.plugin.iceberg.IcebergMetadata.commitTransaction(IcebergMetadata.java:1832)
	... 28 more
```

**Comment by lozbrown** [comment]
@hashhar will these get exported by the open metrics integrations?

What are the metric names here, given that we have the the issues with thread per driver being able to graph that seems useful.

**Comment by hashhar** [comment]
@lozbrown metric names are always generated based on package+class names.

So in this case you can find metrics under `trino.execution.executor.dedicated:name=ThreadPerDriverTaskExecutor`.

**Comment by lozbrown** [comment]
> @lozbrown metric names are always generated based on package+class names.
> 
> 
> 
> So in this case you can find metrics under `trino.execution.executor.dedicated:name=ThreadPerDriverTaskExecutor`.

Thank you.

**Pull request merged**